core/src/main/scala/reftree/diagram/Animation.scala

```diff
@@ -119,6 +107,35 @@ object Animation {
     def iterate(n: Int)(iteration: A ⇒ A) =
       Builder((1 to n).foldLeft(frames)((current, _) ⇒ current :+ iteration(current.last)))
 
+    /** Add more frames by applying the provided iteration function until `predicate` yields true */
+    def iterateUntilAtMost(n: Int)(iteration: (A => A))(predicate: (A => Boolean)) =
+      if (frames.isEmpty) this
+      else
+        Builder(
+          (frames ++ Stream
+            .iterate(frames.last)(iteration)
+            .takeWhileInclusive(x => !predicate(x))
+            .toVector).take(n))
+
+    /** Add more frames by applying the provided iteration function until `predicate` yields true */
+    def iterateUntil(iteration: (A => A))(predicate: (A => Boolean)) =
+      if (frames.isEmpty) this
```

Comment:
Ideally the builder should always have at least one frame, so we should not check that. I can see a few solutions:
* Assume reasonable usage (via `Animation.startWith`) and just ignore the check.
* Make it really hard to construct a builder with 0 frames: make the constructor package-private and the argument private (to disallow `copy`).
* Make it impossible to construct a builder with 0 frames: change the signature to `(frames: Seq[A], lastFrame: A) { def allFrames = frames :+ lastFrame; ... }`